src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -818,6 +836,40 @@ uct_ib_mlx5_devx_open_device(struct ibv_device *ibv_device)
 
 static uct_ib_md_ops_t uct_ib_mlx5_devx_md_ops;
 
+static ucs_status_t uct_ib_mlx5_devx_init_flush_remote(uct_ib_mlx5_md_t *md)
+{
+    intptr_t addr = (intptr_t)md->zero_buf & ~(UCT_IB_MD_MAX_MR_SIZE - 1);
+    /* FW requires indirect atomic MR addr and length to be aligned
+     * to max supported atomic argument size */
+    size_t length = ucs_align_up(ucs_get_page_size() +
+                                 (intptr_t)md->zero_buf - addr,
+                                 ucs_get_page_size());
+    int list_size = ucs_div_round_up(length, UCT_IB_MD_MAX_MR_SIZE);
+    ucs_status_t status;
+
+    status = uct_ib_reg_mr(md->super.pd, md->zero_buf, ucs_get_page_size(),
+                           UCT_IB_MEM_ACCESS_FLAGS, &md->flush_remote_mr, 1);
+    if (status != UCS_OK) {
+        goto out;
+    }
+
+    status = uct_ib_mlx5_devx_reg_ksm_data_addr(md, md->flush_remote_mr,
+                                                (intptr_t)md->zero_buf,
+                                                ucs_get_page_size(),
+                                                0, 0, list_size,
+                                                &md->flush_remote_devx_mr,
+                                                &md->super.flush_remote_rkey);
+    if (status == UCS_OK) {
+        md->super.cap_flags |= UCT_MD_FLAG_FLUSH_REMOTE;
+        ucs_assert(md->super.dummy == 0);
+    } else {
+        uct_ib_dereg_mr(md->flush_remote_mr);
+    }
```

Comment:
add logs to both success and failure
failure is ucs_diag
success is ucs_debug

also, use if (status != UCS_OK) { ... return; }